src/inmanta/validation_type.py

```diff
@@ -26,6 +26,34 @@
 from inmanta.types import PrimitiveTypes
 
 
+def _regex_validator(regex: str) -> pydantic.AfterValidator:
+    """
+    Returns an AfterValidator for regex validation.
+    """
+    # python-re engine can only be selected on model/TypeAdapter level
+    # => add custom validator that delegates to TypeAdapter
+    return pydantic.AfterValidator(
+        pydantic.TypeAdapter(
+            Annotated[
+                str,
+                pydantic.StringConstraints(pattern=regex),
+            ],
+            config=pydantic.ConfigDict(regex_engine="python-re"),
+        ).validate_python
+    )
+
+
+@stable_api
+def regex_string(regex: str) -> object:
```

Comment:
Why not this?
```suggestion
def regex_string(regex: str) -> type:
```
Or even this?
```suggestion
def regex_string(regex: str) -> Type[str]:
```

Reply:
Too bad